tests/components/mikrotik/test_hub.py

```diff
@@ -0,0 +1,181 @@
+"""Test Mikrotik hub."""
+from asynctest import patch
+import librouteros
+import pytest
+
+from homeassistant import config_entries
+from homeassistant.components import mikrotik
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from . import ARP_DATA, DHCP_DATA, MOCK_DATA, WIRELESS_DATA
+
+CONFIG_ENTRY = config_entries.ConfigEntry(
+    version=1,
+    domain=mikrotik.DOMAIN,
+    title="Mikrotik",
+    data=MOCK_DATA,
+    source="test",
+    connection_class=config_entries.CONN_CLASS_LOCAL_POLL,
+    system_options={},
+    options={},
+    entry_id=1,
+)
+
+
+async def setup_mikrotik_entry(hass, **kwargs):
+    """Set up Mikrotik intergation successfully."""
+    support_wireless = kwargs.get("support_wireless", True)
+    dhcp_data = kwargs.get("dhcp_data", DHCP_DATA)
+    wireless_data = kwargs.get("wireless_data", WIRELESS_DATA)
+
+    def mock_command(self, cmd, params=None):
+        if cmd == mikrotik.const.MIKROTIK_SERVICES[mikrotik.const.IS_WIRELESS]:
+            return support_wireless
+        if cmd == mikrotik.const.MIKROTIK_SERVICES[mikrotik.const.DHCP]:
+            return dhcp_data
+        if cmd == mikrotik.const.MIKROTIK_SERVICES[mikrotik.const.WIRELESS]:
+            return wireless_data
+        if cmd == mikrotik.const.MIKROTIK_SERVICES[mikrotik.const.ARP]:
+            return ARP_DATA
+
+    config_entry = CONFIG_ENTRY
+    if "force_dhcp" in kwargs:
+        config_entry.options["force_dhcp"] = True
+
+    if "arp_ping" in kwargs:
+        config_entry.options["arp_ping"] = True
+
+    with patch("librouteros.connect"), patch.object(
+        mikrotik.hub.MikrotikData, "command", new=mock_command
+    ):
+        await mikrotik.async_setup_entry(hass, config_entry)
+        await hass.async_block_till_done()
+        return hass.data[mikrotik.DOMAIN][config_entry.entry_id]
```

Comment:
I'd have this function accept a config entry and return that instance when the set up is done here.